### PR TITLE
Partners summary form

### PR DESCRIPTION
**Pull request by abbiefarr**
# Pull Request Template

## Issue
#298 

Addresses # (issue)

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

When a partner goes to edit a listing, they can toggle between a "Unit Types" and "Individual Units" view. "Individual Units" shows all their units and allows them to add, remove, or edit units. "Unit Types" shows a summary table based on unit type.

image

image

NOTE: This is not a complete feature. Editing the summary table is not possible.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

In the partner portal, edit a listing and toggle between "Unit Types" and "Individual Units"

- [x] Desktop View
- [x] Mobile View

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

**Comment by abbiefarr** [comment]
I wasn't planning on submitting this, I thought I could send screenshots to Safiya that she could show during the property manager meeting. But still wanted to get some eyes on it, I can submit if you think it's useful.

File: sites/partners/src/listings/PaperListingForm/sections/Units.tsx [hunk 1]
```diff
@@ -46,13 +53,28 @@ const FormUnits = ({ units, setUnits, disableUnitsAccordion }: UnitProps) => {
...
+  useEffect(() => {
+    if (unitsSummary !== undefined) {
+      setUnitsSummarized(
+        groupNonReservedAndReservedSummaries(
+          unitsSummary.byNonReservedUnitType,
+          unitsSummary.byReservedType
+        )
+      )
+    }
+  }, [setUnitsSummarized])
```
Comment (willrlin) [inline sites/partners/src/listings/PaperListingForm/sections/Units.tsx:65]: So this sets the initial unitsSummarized once the page loads, right? If there is no unitsSummary passed in as a prop will the table just be blank when you select that option?

**Review by willrlin — APPROVED** [review]

File: ui-components/src/forms/FieldGroup.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ interface FieldGroupProps {
   validation?: Record<string, unknown>
   fieldGroupClassName?: string
   fieldClassName?: string
+  onChange?: () => unknown
```
[the remark this answers is not in the document]
Comment (abbiefarr) [reply ui-components/src/forms/FieldGroup.tsx:24]: Right, they didn't have plumbing to add anything to do onChange. And it's called when you use the selector during editing.

File: sites/partners/src/listings/PaperListingForm/sections/Units.tsx (hunk 1 above)
Comment (abbiefarr) [reply sites/partners/src/listings/PaperListingForm/sections/Units.tsx:65]: unitsSummary is a required prop, so in theory it should always be provided. But the state var defaults to an empty list, so if for some reason unitsSummary is undefined, yes we should just see an empty table.

**Comment by abbiefarr** [comment]
There's a bug in the non-admin property listing page that causes this feature to not work for non-admins. I'll file a ticket for that soon.

**Pull request merged**